pkg/sql/mutations/mutations.go

```diff
@@ -378,6 +383,94 @@ Loop:
 	}
 }
 
+type tableInfo struct {
+	columnNames []tree.Name
+	pkCols      []tree.Name
+}
+
+func indexStoringMutator(rng *rand.Rand, stmts []tree.Statement) ([]tree.Statement, bool) {
+	changed := false
+	tables := map[tree.Name]tableInfo{}
+	getTableInfoFromCreateStatement := func(ct *tree.CreateTable) tableInfo {
+		var columnNames []tree.Name
+		var pkCols []tree.Name
+		for _, def := range ct.Defs {
+			switch ast := def.(type) {
+			case *tree.ColumnTableDef:
+				columnNames = append(columnNames, ast.Name)
+				if ast.PrimaryKey {
+					pkCols = []tree.Name{ast.Name}
+				}
+			case *tree.UniqueConstraintTableDef:
+				if ast.PrimaryKey {
+					for _, elem := range ast.Columns {
+						pkCols = append(pkCols, elem.Column)
+					}
+				}
+			}
+		}
+		return tableInfo{columnNames: columnNames, pkCols: pkCols}
+	}
+	mapFromIndexCols := func(cols []tree.Name) map[tree.Name]struct{} {
+		colMap := map[tree.Name]struct{}{}
+		for _, col := range cols {
+			colMap[col] = struct{}{}
+		}
+		return colMap
+	}
+	generateStoringCols := func(rng *rand.Rand, tableCols []tree.Name, indexCols map[tree.Name]struct{}) []tree.Name {
+		var storingCols []tree.Name
+		for _, col := range tableCols {
+			_, ok := indexCols[col]
+			if ok {
+				continue
+			}
+			if rng.Intn(2) == 0 {
+				storingCols = append(storingCols, col)
+			}
+		}
+		return storingCols
+	}
+	for _, stmt := range stmts {
+		switch ast := stmt.(type) {
+		case *tree.CreateIndex:
+			tableInfo, ok := tables[ast.Table.TableName]
+			if !ok {
+				continue
+			}
+			// If we don't have a storing list, make one with 50% chance.
+			if ast.Storing == nil && rng.Intn(2) == 0 {
+				indexCols := mapFromIndexCols(tableInfo.pkCols)
+				for _, elem := range ast.Columns {
+					indexCols[elem.Column] = struct{}{}
+				}
+				ast.Storing = generateStoringCols(rng, tableInfo.columnNames, indexCols)
+				changed = true
+			}
+		case *tree.CreateTable:
+			// Write down this table for later.
+			tableInfo := getTableInfoFromCreateStatement(ast)
+			tables[ast.Table.TableName] = tableInfo
+			for _, def := range ast.Defs {
+				idx, ok := def.(*tree.IndexTableDef)
```

Comment:
also need to look for tree.UniqueConstraintTableDef

Reply:
are you sure we want to add this for unique constraint def? the unique constraint def is turned into an index later, but we don't really consider it being able to store any other columns

Reply:
I thought this is for things like

```
create table t (a int primary key, b int, c int, unique (b) storing (c));
```

Reply:
TIL